gk/main.c

```diff
@@ -96,6 +487,13 @@ run_gk(struct gk_config *gk_conf)
 	if (!gk_conf)
 		return -1;
 
+	gk_conf->num_lcores = gk_conf->lcore_end_id - gk_conf->lcore_start_id + 1;
+	gk_conf->instances = (struct gk_instance *)rte_calloc(NULL,
+		gk_conf->num_lcores, sizeof(struct gk_instance), 0);
```

Comment:
I think if we do some cleanup here, we may also need to set ``exiting = true`` instead of exiting the program directly, so that the running blocks/instances need to release their allocated resources.